Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -321,18 +358,39 @@ public async Task<string> WalkTree(string treeNodeKey)
                     }
                     finally
                     {
-                        // Always call this callback, whether or not visit node was successful.
-                        await this.Parameters.Callbacks.AfterVisitNode(
+                        if (treeNodeContext_BeforeVisitNode != null)
```

Comment:
>if (treeNodeContext_BeforeVisitNode != null) [](start = 24, length = 44)

If we're just using treeNodeContext_BeforeVisitNode for this null check, I'd prefer to just use the same check here:
if (this.Parameters.CallbacksV2 != null)

Then we don't have to use 2 different names for TreeNodeContext variable. Since they will both live inside separate if statements, we can call them both treeNodeContext.

Reply:
the TreeNodeContext for afterVisitNode, need to get the currentNodeSkipActionContext from the TreeNodeContext for beforeVisitNode. So we  need two variable names, unless we use a local string variable to save the context from beforeVisitNode. Let me know if you think else.

---
In reply to: [505828075](https://github.com/microsoft/Forge/pull/32#discussion_r505828075) [](ancestors = 505828075)